share/completions/brew.fish

```diff
@@ -1,15 +1,56 @@
+function __fish_brew_get_cmd
+  for c in (commandline -opc)
+    if not string match -q -- '-*' $c
+      echo $c
+    end
+  end
+end
+
+function __fish_brew_is_subcommand_services
+  if __fish_brew_using_command services
+    for action in $argv
+      if __fish_brew_using_command $action 3
+        return 0
+      end
+    end
+  end
+
+  return 1
+end
+
 function __fish_brew_needs_command
-  set cmd (commandline -opc)
-  if [ (count $cmd) -eq 1 ]
+  set cmd (__fish_brew_get_cmd)
+  if not set -q cmd[2]
     return 0
   end
   return 1
 end
 
+function __fish_brew_needs_services_action
+  if __fish_brew_using_command services
+    set cmd (__fish_brew_get_cmd)
+    if not set -q cmd[3]
+      return 0
+    end
+  end
+  return 1
+end
+
+function __fish_brew_services
+  brew services list | awk '{if (NR>1) print $1}'
+end
+
 function __fish_brew_using_command
-  set cmd (commandline -opc)
-  if [ (count $cmd) -gt 1 ]
-    if [ $argv[1] = $cmd[2] ]
+  set index 2
+
+  if set -q argv[2]
```

Comment:
@sijad I am sorry to comment on a old PR, but I've used a couple of hours today to read your code and understand what the functions do.  I'm starting to get some sense out of it, but I don't understand this part.

The functions AFAIK only gets passed one argument (argv). E.g.:
>' __fish_brew_using_command sh'

Is it some failsafe?